includes/Core/Util/Activation.php

```diff
@@ -155,9 +155,74 @@ private function get_activation_notice() {
 					ob_start();
 					?>
 					<script type="text/javascript">
-					if( 'undefined' !== typeof sendAnalyticsTrackingEvent ) {
-						sendAnalyticsTrackingEvent( 'plugin_setup', 'plugin_activated' );
-					}
+						if ( 'undefined' !== typeof sendAnalyticsTrackingEvent ) {
+							sendAnalyticsTrackingEvent( 'plugin_setup', 'plugin_activated' );
+						}
+
+						jQuery( document ).ready( function( $ ) {
+							var trackingScriptPresent = !! googlesitekit.admin.trackingOptIn;
+
+							if ( googlesitekit.admin.trackingOptIn ) {
+								$( '#opt-in' ).prop( 'checked', googlesitekit.admin.trackingOptIn );
+							}
+							if ( googlesitekit.admin.proxySetupURL ) {
+								$( '#start-setup-link' ).attr( 'href', googlesitekit.admin.proxySetupURL );
+							}
+
+							$( '#start-setup-link' ).on( 'click' , function() {
+								if ( 'undefined' !== typeof sendAnalyticsTrackingEvent ) {
+									sendAnalyticsTrackingEvent( 'plugin_setup', 'goto_sitekit' );
+								}
+							} );
+
+							$('#opt-in').on( 'change' , function( event ) {
+								if ( event.target.disabled ) {
+									event.preventDefault();
+									return;
+								}
+
+								var checked = event.target.checked;
+
```

Comment:
For the lack of state management, we need to update `window.googlesitekitTrackingEnabled` here accordingly because the `sendAnalyticsTrackingEvent()` function loaded in the `Tracking` class relies on it.

Reply:
Actually, I realised that `optin.js` was never disabling `window.googlesitekitTrackingEnabled` when unchecked. So I've added that here as well as to `<OptIn />`.